src/components/TodoFormList.tsx

```diff
@@ -0,0 +1,192 @@
+  const { id, completed, title } = todo;
+
+  const [isEditTodo, setIsEditTodo] = useState<{
+    [id: number]: boolean;
+  } | null>(null);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editTitle, setEditTitle] = useState('');
+  const inputRefFocus = useRef<HTMLInputElement>(null);
+  const returnEditionRef = useRef<(e: KeyboardEvent) => void>(() => {});
+
+  const { todos, inputHeaderRef } = useGlobalState();
+  const dispatch = useDispatch();
+
+  const deleteTodo = (idToDelete: number) => {
+    deleteTodoFromStorage(idToDelete);
+    dispatch({ type: 'deleteTodo', payload: id });
+
+    if (inputHeaderRef?.current) {
+      inputHeaderRef.current.focus();
+    }
+  };
+
+  const updateTodo = (idToUpdate: number, data: Omit<Todo, 'id'>) => {
+    patchTodoFromStorage(idToUpdate, data);
+    dispatch({ type: 'patchTodo', payload: { id, data } });
+  };
+
+  const toggleCompleted = (
+    idToToggle: number,
+    completedToToggle: boolean,
+    titleToToggle: string,
+  ) => {
+    const data = {
+      completed: !completedToToggle,
+      title: titleToToggle,
+    };
+
+    updateTodo(idToToggle, data);
+  };
+
+  const updateTitleTodo = (
+    idToUpdate: number,
+    completedToUpdate: boolean,
+    titleToUpdate: string,
+  ) => {
+    const data = {
+      completed: completedToUpdate,
+      title: titleToUpdate,
+    };
+
+    updateTodo(idToUpdate, data);
+  };
+
+  useEffect(() => {
+    const returnEdition = (e: KeyboardEvent) => {
+      if (e.code === 'Escape') {
+        setEditTitle('');
+        setIsEditTodo(null);
+        setIsEditing(false);
+      }
+    };
+
+    returnEditionRef.current = returnEdition;
+
+    if (isEditTodo) {
+      window.addEventListener('keyup', returnEditionRef.current);
+    }
+
+    return () => {
+      if (returnEditionRef.current) {
+        window.removeEventListener('keyup', returnEditionRef.current);
+      }
+    };
+  }, [isEditTodo]);
+
+  const handleEditTodo = (idTodo: number, titleTodo: string) => {
+    setIsEditTodo(null);
+    setIsEditTodo({ [idTodo]: true });
+    setEditTitle(titleTodo);
+    setIsEditing(true);
+  };
+
+  useEffect(() => {
+    if (isEditing && inputRefFocus.current) {
+      inputRefFocus.current.focus();
+    }
+  }, [isEditing]);
+
+  const handleInputTitle = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEditTitle(e.currentTarget.value);
+  };
+
+  const submitEditTitle = (idTodo: number) => {
+    const index = todos.findIndex(t => idTodo === t.id);
+
+    if (todos[index].title === editTitle) {
+      setEditTitle('');
+      setIsEditTodo(null);
+      setIsEditing(false);
+
+      return;
+    }
+
+    if (editTitle.trim().length === 0) {
+      deleteTodo(idTodo);
+
+      return;
+    }
+
+    updateTitleTodo(idTodo, todos[index].completed, editTitle.trim());
+    setEditTitle('');
+    setIsEditTodo(null);
+    setIsEditing(false);
+  };
+
+  const handleBlur = (idTodo: number) => {
```

Comment:
redundant wrapper - just use `submitEditTitle` everywhere instead of `handleBlur`